sparkplug-core/src/clojure/sparkplug/core.clj

```diff
@@ -287,16 +287,15 @@
 (defn intersection
   "Construct an RDD representing the intersection of elements which are in both
   RDDs."
-  ^JavaRDD
-  [^JavaRDD rdd1 ^JavaRDD rdd2]
+  [rdd1 rdd2]
```

Comment:
Since the rest of the funtions are type-hinted, it would probably be worth a quick comment in the spots where you've removed the hints noting that the lack is intentional.

Reply:
Good suggestion, I've added some comments.